File: project/.sdk/tm/rb/utility/prepare_path.rb
# ProjectName SDK utility: prepare_path
require_relative 'struct/voxgig_struct'
module ProjectNameUtilities
  PreparePath = ->(ctx) {
    point = ctx.point
    parts = []
    if point
      p = VoxgigStruct.getprop(point, "parts")
      parts = p if p.is_a?(Array)
    end
    VoxgigStruct.join(parts, "/", true)
  }
end

File: ts/project/.sdk/tm/rust/utility/prepare_path.rs
use std::rc::Rc;

use crate::core::context::Context;
use crate::core::helpers::getp;
use crate::utility::voxgigstruct as vs;
use crate::utility::voxgigstruct::Value;

pub fn prepare_path_util(ctx: &Rc<Context>) -> String {
    let point = ctx.point.borrow().clone();

    let parts = match getp(&point, "parts") {
        Value::List(l) => Value::List(l),
        _ => Value::empty_list(),
    };

    vs::join(&parts, Some("/"), true)
}
